internal/servers/controller/handler.go

```diff
@@ -176,73 +147,104 @@ func handleGrpcGateway(c *Controller, props HandlerProperties) (http.Handler, er
 		}
-		services.RegisterRoleServiceServer(c.gatewayServer, rs)
-		if err := services.RegisterRoleServiceHandlerFromEndpoint(ctx, c.gatewayMux, gatewayTarget, dialOptions); err != nil {
-			return nil, fmt.Errorf("failed to register role service handler: %w", err)
-		}
+		services.RegisterRoleServiceServer(s, rs)
 	}
 	if _, ok := currentServices[services.SessionService_ServiceDesc.ServiceName]; !ok {
 		ss, err := sessions.NewService(c.SessionRepoFn, c.IamRepoFn)
 		if err != nil {
 			return nil, fmt.Errorf("failed to create session handler service: %w", err)
 		}
-		services.RegisterSessionServiceServer(c.gatewayServer, ss)
-		if err := services.RegisterSessionServiceHandlerFromEndpoint(ctx, c.gatewayMux, gatewayTarget, dialOptions); err != nil {
-			return nil, fmt.Errorf("failed to register session service handler: %w", err)
-		}
+		services.RegisterSessionServiceServer(s, ss)
 	}
 	if _, ok := currentServices[services.ManagedGroupService_ServiceDesc.ServiceName]; !ok {
 		mgs, err := managed_groups.NewService(c.OidcRepoFn)
 		if err != nil {
 			return nil, fmt.Errorf("failed to create managed groups handler service: %w", err)
 		}
-		services.RegisterManagedGroupServiceServer(c.gatewayServer, mgs)
-		if err := services.RegisterManagedGroupServiceHandlerFromEndpoint(ctx, c.gatewayMux, gatewayTarget, dialOptions); err != nil {
-			return nil, fmt.Errorf("failed to register managed groups service handler: %w", err)
-		}
+		services.RegisterManagedGroupServiceServer(s, mgs)
 	}
 	if _, ok := currentServices[services.CredentialStoreService_ServiceDesc.ServiceName]; !ok {
 		cs, err := credentialstores.NewService(c.VaultCredentialRepoFn, c.IamRepoFn)
 		if err != nil {
 			return nil, fmt.Errorf("failed to create credential store handler service: %w", err)
 		}
-		services.RegisterCredentialStoreServiceServer(c.gatewayServer, cs)
-		if err := services.RegisterCredentialStoreServiceHandlerFromEndpoint(ctx, c.gatewayMux, gatewayTarget, dialOptions); err != nil {
-			return nil, fmt.Errorf("failed to register credential store service handler: %w", err)
-		}
+		services.RegisterCredentialStoreServiceServer(s, cs)
 	}
 	if _, ok := currentServices[services.CredentialLibraryService_ServiceDesc.ServiceName]; !ok {
 		cl, err := credentiallibraries.NewService(c.VaultCredentialRepoFn, c.IamRepoFn)
 		if err != nil {
 			return nil, fmt.Errorf("failed to create credential library handler service: %w", err)
 		}
-		services.RegisterCredentialLibraryServiceServer(c.gatewayServer, cl)
-		if err := services.RegisterCredentialLibraryServiceHandlerFromEndpoint(ctx, c.gatewayMux, gatewayTarget, dialOptions); err != nil {
-			return nil, fmt.Errorf("failed to register credential library service handler: %w", err)
-		}
+		services.RegisterCredentialLibraryServiceServer(s, cl)
+	}
+
+	return s, nil
+}
+
+func registerGrpcGatewayEndpoints(ctx context.Context, gwMux *runtime.ServeMux, dialOptions ...grpc.DialOption) (*runtime.ServeMux, error) {
```

Comment:
https://github.com/hashicorp/boundary/pull/1874/files#r811054909 same situation

Reply:
Also went with 1. Addressed in [64a5d98](https://github.com/hashicorp/boundary/pull/1874/commits/64a5d98342d9db493fdcc167593d6a1397f2ca98)